src/hotspot/share/opto/compile.cpp

```diff
@@ -4022,6 +4025,14 @@ bool Compile::final_graph_reshaping() {
         record_method_not_compilable("malformed control flow");
         return true;            // Not all targets reachable!
       }
+    } else if (n->is_PCTable() && n->in(0) && n->in(0)->in(0) && n->in(0)->in(0)->is_Call()) {
+      CallNode* call = n->in(0)->in(0)->as_Call();
+      if (call->entry_point() == OptoRuntime::new_array_Java() ||
+          call->entry_point() == OptoRuntime::new_array_nozero_Java()) {
+        assert(call->is_CallStaticJava(), "static call expected");
+        assert(call->req() == call->jvms()->endoff() + 1, "missing extra input");
+        call->del_req(call->req()-1); // valid length test useless now
```

Comment:
How you remove graph (test) attached to ValidLengthTest input? How you know that `call` node still has this input and it is `call->req() -1`?

Reply:
You're right that the subgraph at ValidLengthTest goes dead and is not properly removed. I pushed a new change to fix that.
The ValidLengthTest input from the AllocateArrayNode is always moved to the call so there's no reason it shouldn't still be there.